pkg/routeagent_driver/handlers/ovn/submarinerNonGWRouteHandler.go

```diff
@@ -0,0 +1,189 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ovn
+
+import (
+	"context"
+	"os"
+	"sync"
+
+	"github.com/pkg/errors"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	submarinerClientset "github.com/submariner-io/submariner/pkg/client/clientset/versioned"
+	"github.com/submariner-io/submariner/pkg/event"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+)
+
+type NonGatewayRouteHandler struct {
+	event.HandlerBase
+	mutex           sync.Mutex
+	smClient        submarinerClientset.Interface
+	localEndpoint   *submarinerv1.Endpoint
+	remoteEndpoints map[string]*submarinerv1.Endpoint
+	k8sClientSet    clientset.Interface
+	isGateway       bool
+	transitSwitchIP string
+}
+
+func NewNonGatewayRouteHandler(smClientSet submarinerClientset.Interface, k8sClientset *clientset.Clientset) *NonGatewayRouteHandler {
+	// We'll panic if env is nil, this is intentional
+	return &NonGatewayRouteHandler{
+		smClient:        smClientSet,
+		remoteEndpoints: map[string]*submarinerv1.Endpoint{},
+		k8sClientSet:    k8sClientset,
+	}
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) Init() error {
+	logger.Infof("Starting NonGatewayRouteHandler")
+
+	nodeName, ok := os.LookupEnv("NODE_NAME")
+	if !ok {
+		return errors.New("error getting the Node name")
+	}
+
+	node, err := nonGatewayRouteHandler.k8sClientSet.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "error getting the g/w node: %q", nodeName)
+	}
+
+	annotations := node.GetAnnotations()
+
+	transitSwitchIP, ok := annotations["k8s.ovn.org/node-transit-switch-port-ifaddr"]
+	if !ok {
+		logger.Infof("No transit switch IP configured")
+		return nil
+	}
+
+	nonGatewayRouteHandler.transitSwitchIP = transitSwitchIP
```

Comment:
What happened if transitSwitchIP will be changed?

Reply:
it is a topology change right ? we don't support it right now I think, like if ovn-k8s-mp0 interface IP change.

Reply:
If we plan to support it in phase-2, maybe it is worth adding a TODO note?

Reply:
Added TODO